### PR TITLE
vdk-lineage: introducing POC (pre-alpha) implementation

### PR DESCRIPTION
**Pull request by tozka**
VDK Lineage plugin provides lineage data (input data -> job -> output
data) information and send it to pre-configured
destination.

At POC level currently (Pre-Alpha release).
It collects lineage information for each job run and for each executed
query.
Query execution is currently before it's executed (so not query status
is logged) as we lack the necessary hooks.

There are plenty of TODOs in the code that need to be addressed before
moving maturity level and add more tests.

Testing Done: there are some tests and also manually:
```
vdk marquez-server --start
vdk run some-job
vdk marquez-server --stop
```
and saw lineage in marquz ui

Signed-off-by: Antoni Ivanov <aivanov@vmware.com>

### CODE REVIEW

File: projects/vdk-plugins/vdk-lineage/.plugin-ci.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+image: "python:3.7"
+
+.build-vdk-lineage: # change name here to the name of the plugin
```
Comment (gageorgiev) [inline projects/vdk-plugins/vdk-lineage/.plugin-ci.yml:6]: delete this comment - and the subsequent ones - since the name has been changed

File: projects/vdk-plugins/vdk-lineage/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# VDK Lineage
+
+VDK Lineage plugin provides lineage data (input data -> job -> output data) information and send it to pre-configured
```
Comment (gageorgiev) [inline projects/vdk-plugins/vdk-lineage/README.md:3]:
```suggestion
VDK Lineage plugin provides lineage data (input data -> job -> output data) information and send it to a pre-configured
```

File: projects/vdk-plugins/vdk-lineage/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+destination.
+
+At POC level currently. It collect lineage information for each job run and for each executed query.
+Query execution is currently before it's executed (so not query status is logged).
```
Comment (gageorgiev) [inline projects/vdk-plugins/vdk-lineage/README.md:7]: I'm confused as to what this sentence means. Maybe rephrase it?

File: projects/vdk-plugins/vdk-lineage/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
+openlineage-integration-common
+openlineage-python
+pytest
+pytest-httpserver
+sqllineage
+sqlparse
+vdk-core
```
Comment (gageorgiev) [inline projects/vdk-plugins/vdk-lineage/requirements.txt:7]: Isn't the requirements.txt file only for testing dependencies, given that the rest are already listed in the setup.py file? It seems reasonable to me to not have them in two places in case the deps change at some point in the future.

File: projects/vdk-plugins/vdk-lineage/src/vdk/api/lineage/sql/data.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+from typing import List
+
+
+@dataclass(frozen=True)
+class LineageTable:
+    """
+    Defines the LineageTable object contract
+
+    Attributes
+    catalog: str
```
[the remark this answers is not in the document]
Comment (tozka) [reply projects/vdk-plugins/vdk-lineage/src/vdk/api/lineage/sql/data.py:14]: Fixed by changing the type hint to be optional which is better.

File: projects/vdk-plugins/vdk-lineage/requirements.txt (hunk 4 above)
Comment (tozka) [reply projects/vdk-plugins/vdk-lineage/requirements.txt]: IntelliJ recognizes requrements.txt and installs them automatically. But having a clear separation may be better so I removed them.

File: projects/vdk-plugins/vdk-lineage/README.md (hunk 3 above)
Comment (tozka) [reply projects/vdk-plugins/vdk-lineage/README.md]: I re-wrote the README entirely . Hopefully it's now clearer/better.

File: projects/vdk-plugins/vdk-lineage/src/vdk/plugin/marquez/marquez_server.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import webbrowser
+from typing import List
+
+log = logging.getLogger(__name__)
+
+
+def execute(command, success_codes=(0,), env=None):
+    """Run a shell command."""
+    try:
+        output = subprocess.check_output(command, stderr=subprocess.STDOUT, shell=True)
```
[the remark this answers is not in the document]
Comment (tozka) [reply projects/vdk-plugins/vdk-lineage/src/vdk/plugin/marquez/marquez_server.py:16]: Currently, the command is constructed without user input so there's no chance of shell injection. 

But it's a simple fix and all equal it's always better to have prevention for potential future changes. 
Thanks for the hint!

[the remark this answers is not in the document]
Comment (tozka) [reply projects/vdk-plugins/vdk-lineage/src/vdk/plugin/marquez/marquez_server.py:16]: I will have to fix that in a subsequent change. It's not working for me now and as stated there's no user input.
